vaccine/vaccine_reg_ussd.py

```diff
@@ -136,12 +134,33 @@ async def state_terms_and_conditions_2(self):
         )
 
     async def state_terms_and_conditions_3(self):
+        def next_state():
+            if (
+                self.user.answers.get("state_identification_type")
+                and self.user.answers.get("state_identification_number")
+                and self.user.answers.get("state_dob_year")
+                and self.user.answers.get("state_dob_month")
+                and self.user.answers.get("state_dob_day")
+                and self.user.answers.get("state_gender")
+            ):
+                return "state_first_name"
+            elif (
+                self.user.answers.get("state_identification_type")
+                and self.user.answers.get("state_identification_number")
+                and "state_dob_year" not in self.user.answers
+                and self.user.answers.get("state_dob_month")
+                and self.user.answers.get("state_dob_day")
+                and self.user.answers.get("state_gender")
+            ):
+                return "state_dob_year"
+            return "state_identification_type"
```

Comment:
Instead of having this large if statement (which might break if the order of the states changes), what if we just have inside `state_identification_type`
```
if self.user.answers.get("state_identification_type"):
    return await self.go_to_state("state_identification_number")
```
And in `state_identification_number` we have
```
if self.user.answers.get("state_identification_number"):
    return await self.go_to_state(next_state)
```

That way, it will just skip through the `identification_type` and `identification_number` states if we already have a value for them, and reuse the "skip DoB if we already have it" logic that is already there.